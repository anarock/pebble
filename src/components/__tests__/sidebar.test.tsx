import * as React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import SideBar from "../Sidebar";
import sinon from "sinon";

function noop() {}

describe("Sidebar snapshot tests", () => {
  test("open", () => {
    const clock = sinon.useFakeTimers();

    const sidebar = renderer.create(
      <SideBar isOpen width={200} onClose={noop}>
        Sidebar
      </SideBar>
    );

    clock.tick(2000);

    const tree = sidebar.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("closed", () => {
    const clock = sinon.useFakeTimers();

    const sidebar = renderer.create(
      <SideBar isOpen={false} width={200} onClose={noop}>
        Sidebar
      </SideBar>
    );

    clock.tick(2000);

    const tree = sidebar.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Sidebar func tests", () => {
  test("onClose should be called", () => {
    const spy = sinon.spy();
    const sidebar = mount(
      <SideBar onClose={spy} isOpen width={300}>
        Sidebar
      </SideBar>
    );
    sidebar
      .find(".pi.pi-close")
      .at(0)
      .simulate("click");
    expect(spy.calledOnce).toBeTruthy();
  });
});
