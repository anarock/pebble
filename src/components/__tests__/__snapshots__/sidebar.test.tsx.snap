// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar snapshot tests closed 1`] = `
.emotion-0 {
  width: 200px;
  -webkit-transform: translateX(200px);
  -ms-transform: translateX(200px);
  transform: translateX(200px);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(.64,.09,.08,1);
  -webkit-transition: transform 200ms cubic-bezier(.64,.09,.08,1);
  transition: transform 200ms cubic-bezier(.64,.09,.08,1);
  background-color: #ffffff;
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  will-change: transform;
  z-index: 99;
  webkit-overflow-scrolling: touch;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
}

@media (max-width:800px) {
  .emotion-0 {
    width: 100%;
    padding-top: 50px;
  }
}

<div>
  <div
    className="emotion-0"
  >
    <div
      style={
        Object {
          "height": "100vh",
          "overflowY": "scroll",
        }
      }
    >
      Sidebar
    </div>
  </div>
</div>
`;

exports[`Sidebar snapshot tests open 1`] = `
Array [
  .emotion-0 {
  background-color: rgba(16,23,33,0.3);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 99;
}

<div
    className="emotion-0"
    style={
      Object {
        "opacity": 1,
      }
    }
  />,
  .emotion-0 {
  background-color: #ffffff;
  border-radius: 3px;
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin-left: -60px;
  margin-top: 20px;
  position: absolute;
}

.emotion-0 i {
  font-size: 14px;
  color: #101721;
}

@media (max-width:800px) {
  .emotion-0 {
    left: 80px;
    margin-top: -45px;
    border-radius: 0;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
    padding: 0 20px;
  }

  .emotion-0 i {
    float: right;
    color: #6B7785;
    font-size: 14px;
  }
}

.emotion-1 {
  width: 200px;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(.64,.09,.08,1);
  -webkit-transition: transform 200ms cubic-bezier(.64,.09,.08,1);
  transition: transform 200ms cubic-bezier(.64,.09,.08,1);
  background-color: #ffffff;
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  will-change: transform;
  z-index: 99;
  webkit-overflow-scrolling: touch;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

@media (max-width:800px) {
  .emotion-1 {
    width: 100%;
    padding-top: 50px;
  }
}

<div>
    <div
      className="emotion-1"
    >
      <div
        className="emotion-0"
        onClick={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <i
          className="pi pi-close"
        />
        <canvas
          height={0}
          onDragOver={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          style={
            Object {
              "borderRadius": "inherit",
              "height": "100%",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
          width={0}
        />
      </div>
      <div
        style={
          Object {
            "height": "100vh",
            "overflowY": "scroll",
          }
        }
      >
        Sidebar
      </div>
    </div>
  </div>,
]
`;
